src/main/mondrian/rolap/SqlTupleReader.java

```diff
@@ -841,7 +841,7 @@ private void savePartialResult(List<List<RolapMember>> partialResult) {
                 if (fullyJoiningBaseCubes.size() > 1) {
                     for (int i = 0; i < types.size(); i++) {
                         unionQuery.addOrderBy(
-                            i + 1 + "",
+                            "c" + i,
                             true,
```

Comment:
We probably want to make this change conditional on whether the dialect has requiresOrderByAlias=true.  It's possible (likely) that some databases which don't require aliases will fail with the new UNION structure if we don't have a numbered order by list.

And to confirm-- you've tried older versions of MySQL with this change to the UNION sql?  (and full test suite?)